src/fhirtypes/StructureDefinition.ts

```diff
@@ -148,35 +148,37 @@ export class StructureDefinition {
     let fhirPathString = this.type;
     let matchingElements = this.elements;
     let newMatchingElements: ElementDefinition[] = [];
+    let unfoldedElements: ElementDefinition[] = [];
     // Iterate over the path, filtering out elements that do not match
     for (const pathPart of parsedPath) {
       // Add the next part to the path, and see if we have matches on it
       fhirPathString += `.${pathPart.base}`;
       newMatchingElements = matchingElements.filter(e => e.path.startsWith(fhirPathString));
 
+      if (newMatchingElements.length === 0 && matchingElements.length === 1) {
+        // If there was previously only one match,
+        // we want to unfold that match and dig deeper into it
+        unfoldedElements = matchingElements[0].unfold(resolve);
+        if (unfoldedElements.length > 0) {
+          // Only get the children that match our path
+          newMatchingElements = unfoldedElements.filter(e => e.path.startsWith(fhirPathString));
+        }
+      }
+
       if (newMatchingElements.length === 0) {
-        // If we fail to find any matches, first try to find the appropriate [x] element
+        // If we fail to find any matches, try to find the appropriate [x] element
+        // from previous matches or from newly unfolded elements
         // Ex: valueString -> value[x]
-        const matchingSlice = this.sliceMatchingValueX(fhirPathString, matchingElements);
+        const matchingSlice = this.sliceMatchingValueX(fhirPathString, [
+          ...matchingElements,
+          ...unfoldedElements
+        ]);
         if (matchingSlice) {
           newMatchingElements.push(matchingSlice, ...matchingSlice.children());
           fhirPathString = matchingSlice.path;
         }
       }
 
-      // TODO: If path is A.B.C, and we unfold B, but C is invalid, the unfolded
-      // elements are still on the structDef. We may want to change this to remove the elements
-      // upon error
```

Comment:
It seems we dropped this TODO comment.  Was that intentional?  It seems to me that it probably still applies?